### ISSUE TITLE
No disconnect action is being taken

### ISSUE THREAD

**Issue by achaloyan** [unispeech/unimrcp#162]
Originally reported on Google Code with ID 163

```
What steps will reproduce the problem?
1. start unimrcp server (use sip over tcp in config)
2. start umc.exe and give run recog
3. kill umc process at some random point

What version of the product are you using? On what operating system?
unimrcp 1.1.0. the problem probably continues in trunk.
windows7 64 bit.

Please provide any additional information below.
I see that neither channel close nor channel destroyed when the client app exits (crashes
vs.). In fact we discovered that in our stress tests. We exit the umc client at random
points  and observe if unimrcpserver.exe is  in a stable state. Further analysis showed
that the following function has not been implemented:

apt_bool_t mrcp_server_on_disconnect(mrcp_channel_t *channel)
{
    /* to be processed */
    return TRUE;
}

I think this is probably related to our problem.
mrcp_server_on_engine_channel_close seems to start a series of calls which may lead
to calling mrcp_server_session_terminate_process in the end. Could simply calling mrcp_server_on_engine_channel_close
from this function possible let us solve the leaking connection problem and let the
corresponding channel close/destroy function be called. 

Could you please direct us on how to process server disconnect function. 

```

Reported by `mert.buyuktuncay@sestek.com` on 2014-02-24 09:58:08

**Comment by achaloyan** [comment]
Fixed in https://github.com/unispeech/unimrcp/commit/03532d2280409f3dc1e00f96e11942865ed9c422

**Comment by amasse76** [comment]
Hi Arsen,

I ran into the same issue with a previous version of unimrcp. I updated my current code with your modification ( 03532d2 + 34308e2 ) but the behaviour seems strange to me. The code sends a nua_bye() but I need to wait 32 seconds to receive a "408 Request Timeout" and generate the cleaning mechanism.

Is it the expecter behavior ?

Thanks
Anthony

2017-10-23 16:31:23:805652 04856 [NOTICE] Initiate Session Termination <3100919b05e6c947>

2017-10-23 16:31:55:806852 05508 [INFO]   Receive SIP Event [nua_r_bye] Status 408 Request Timeout
2017-10-23 16:31:55:806852 05508 [INFO]   Receive SIP Event [nua_i_state] Status 408 to BYE
...
2017-10-23 16:31:55:816853 04856 [NOTICE] Destroy Session <3100919b05e6c947>
